pkg/performanceprofile/controller/performanceprofile/hypershift/status/status.go

```diff
@@ -0,0 +1,123 @@
+package status
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+
+	performancev2 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/performanceprofile/v2"
+	tunedv1 "github.com/openshift/cluster-node-tuning-operator/pkg/apis/tuned/v1"
+	handler "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/hypershift/components"
+	hypershiftconsts "github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/hypershift/consts"
+	"github.com/openshift/cluster-node-tuning-operator/pkg/performanceprofile/controller/performanceprofile/status"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+)
+
+const (
+	ppStatusConfigMapPrefix = "status"
+)
+
+func getStatusConfigMapName(instanceName string) string {
+	return fmt.Sprintf("%s-%s", ppStatusConfigMapPrefix, instanceName)
+}
+
+func createOrUpdateStatusConfigMap(ctx context.Context, cli client.Client, cm *corev1.ConfigMap, profileName string, conditions []conditionsv1.Condition) error {
+	prevStatus, prevStatusFound, err := getPreviousStatusFrom(ctx, cli, &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cm.Name,
+			Namespace: cm.Namespace,
+		},
+	})
+	if err != nil {
+		return err
+	}
+	npName := cm.Labels[hypershiftconsts.NodePoolNameLabel]
+	updatedStatus := status.CalculateUpdatedStatus(prevStatus, profileName, npName, conditions)
+	if updatedStatus == nil {
+		return nil
+	}
+	encodedStatus, encodeErr := yaml.Marshal(updatedStatus)
+	if encodeErr != nil {
+		return encodeErr
+	}
+	cm.Data = map[string]string{
+		hypershiftconsts.PerformanceProfileStatusKey: string(encodedStatus),
+	}
+
+	key := client.ObjectKeyFromObject(cm)
+	if prevStatusFound {
+		klog.InfoS("Updating status", "ConfigMap", key.String())
+		if err := cli.Update(ctx, cm); err != nil {
+			return fmt.Errorf("failed to update status ConfigMap %q: %w", key.String(), err)
+		}
+		klog.InfoS("Status updated ", "ConfigMap", key.String())
+	} else {
```

Comment:
minor/nit: please remove the `else` block and align to the left

Reply:
Done. Thank you